src/Estimators/src/SubModelKinDynWrapper.cpp

```diff
@@ -0,0 +1,328 @@
+    m_baseVelocity.setZero();
+
+    // Initialize attributes with correct sizes
+    m_massMatrix.resize(6+m_subModel.getModel().getNrOfDOFs(), 6+m_subModel.getModel().getNrOfDOFs());
+    m_massMatrix.setZero();
+
+    m_genForces.resize(6+m_subModel.getModel().getNrOfDOFs());
+    m_genForces.setZero();
+
+    for (int idx = 0; idx < m_subModel.getFTList().size(); idx++)
+    {
+        Eigen::MatrixXd tempJacobian(6, 6 + m_numOfJoints);
+        tempJacobian.setZero();
+        m_jacFTList[m_subModel.getFTList()[idx].name] = std::move(tempJacobian);
+    }
+
+    for (int idx = 0; idx < m_subModel.getAccelerometerList().size(); idx++)
+    {
+        Eigen::MatrixXd tempJacobian(6, 6 + m_numOfJoints);
+        tempJacobian.setZero();
+        m_jacAccList[m_subModel.getAccelerometerList()[idx].name] = std::move(tempJacobian);
+
+        Eigen::VectorXd tempAcceleration(6);
+        tempAcceleration.setZero();
+        m_dJnuList[m_subModel.getAccelerometerList()[idx].name] = std::move(tempAcceleration);
+
+        Eigen::Matrix3d rotMatrix;
+        rotMatrix.setZero();
+        m_accRworldList[m_subModel.getAccelerometerList()[idx].name] = std::move(rotMatrix);
+    }
+
+    for (int idx = 0; idx < m_subModel.getGyroscopeList().size(); idx++)
+    {
+        Eigen::MatrixXd tempJacobian(6, 6 + m_numOfJoints);
+        tempJacobian.setZero();
+        m_jacGyroList[m_subModel.getGyroscopeList()[idx].name] = std::move(tempJacobian);
+    }
+
+    for (int idx = 0; idx < m_subModel.getExternalContactList().size(); idx++)
+    {
+        Eigen::MatrixXd tempJacobian(6, 6 + m_numOfJoints);
+        tempJacobian.setZero();
+        m_jacContactList[m_subModel.getExternalContactList()[idx]] = std::move(tempJacobian);
+    }
+
+    return true;
+}
+
+bool RDE::SubModelKinDynWrapper::updateKinDynState(
+    std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel)
+{
+    constexpr auto logPrefix = "[BipedalLocomotion::Estimators::SubModelKinDynWrapper::"
+                               "updateKinDynState]";
+
+    // Get model base position and velocity
+    const std::string baseFrameName = m_kinDyn.getFloatingBase();
+
+    // Get world transform for the base frame
+    iDynTree::Transform worldTBase = kinDynFullModel->getWorldTransform(baseFrameName);
+
+    // Get base velocity
+    iDynTree::Twist baseVelocity = kinDynFullModel->getFrameVel(baseFrameName);
+
+    // Get position and velocity of the joints in the full model
+    iDynTree::VectorDynSize jointPositionModel(kinDynFullModel->getNrOfDegreesOfFreedom());
+    iDynTree::VectorDynSize jointVelocityModel(kinDynFullModel->getNrOfDegreesOfFreedom());
+    iDynTree::Vector3 worldGravity;
+    kinDynFullModel->getRobotState(jointPositionModel, jointVelocityModel, worldGravity);
+
+    // Get positions and velocities of joints of the sub model
+    iDynTree::VectorDynSize qj(m_subModel.getModel().getNrOfDOFs()),
+        dqj(m_subModel.getModel().getNrOfDOFs());
```

Comment:
Apologies for not noticing it before, but also ``jointPositionModel``, ``jointVelocityModel`` and ``qj`` require dynamic memory allocation